### ISSUE TITLE
Meilisearch automatically starts with a master key when a master key isn't specified

### ISSUE THREAD

**Issue by oluademola** [meilisearch/meilisearch#3733]
### Discussed in https://github.com/meilisearch/meilisearch/discussions/3712

<div type='discussions-op-text'>

<sup>Originally posted by **coolb0y** April 30, 2023</sup>
I downloaded windows binary of meilisearch and double clicked it It should not have authentication start own its own why It has got authentication enabled (I am using windows RDP to test Meilisearch windows 10)

**Description**
- Meilisearch automatically starts with a master key when a master key isn't specified. 

**Expected behavior**
- Meilisearch should be unprotected by default. 

**Screenshots or Logs**
![Screenshot 2023-05-08 100923](https://user-images.githubusercontent.com/101267639/236788508-9cbe8adf-32f1-41be-8eae-88dd7eae52dc.png)

**Environment**
- OS and version: Windows 10 and 11
- Meilisearch version:  v.1.1.1

**Comment by Kerollmops** [comment]
Hey @oluademola,

I am maybe wrong, but shouldn't the API key be prefixed by the `Bearer` word as the message explains?

Have a nice day 🏮

**Comment by oluademola** [comment]
Hello @Kerollmops. Sorry for that. My example should have included it, but the issue is more than that. I had linked this issue to the discussion because of the first problem. The problem is not with the bearer as you rightly pointed out but is due to meilisearch flagging the master key as invalid when it is indeed valid. Here is an example using `GtDQyOk3mv2uj5hTcWq3l6EuhvpnSAgtt5zQkLaSTYQ` as my master key:

![Screenshot 2023-05-10 102725](https://github.com/meilisearch/meilisearch/assets/101267639/157fdc86-8d2b-4261-9290-014b03b383fa)
---
![Screenshot 2023-05-10 102814](https://github.com/meilisearch/meilisearch/assets/101267639/5fd30045-f918-4668-a87c-966e4aa1474a)

**Comment by curquiza** [comment]
Thanks for the report @oluademola 
Can you confirm the same manipulation work on Linux? Is it only on Windows?

**Comment by oluademola** [comment]
Hello @curquiza,

After taking a closer look at the issue, I realized I was hitting the invalid API key error because I had a meilisearch background service running. Typically, I would expect an address already in use error, just Linux or macOS, when launching a new instance with the same port, but that wasn’t the case. After stopping the background service, the authentication works fine. Sorry for the false alarm on that one; the problem wasn’t more than just including the bearer in the authorization header. However, the first issue persists as meilisearch still runs with some unknown master key when I launch it without a master key.

![Screenshot 2023-05-10 142333](https://github.com/meilisearch/meilisearch/assets/101267639/a3f04f99-cb85-45c9-b699-6aa9ed58707c)

PS: It happens only on windows.

**Comment by irevoire** [comment]
Hey @oluademola, can you confirm one last thing;

> However, the first issue persists as meilisearch still runs with some unknown master key when I launch it without a master key.

If you try to access another resource than the `key` one (which is a special case since you can't access it without a master key), do you still get the error?
If you run this command, for example:
```bash
curl -X GET http://localhost:7700/indexes -H Authorization:YCggmAN3K_ECwy10qKHa1UPGSoa99ZGgYQMVPbuOBCo
```

**Comment by oluademola** [comment]
Hello @irevoire. There is no issue when I launch it with a master key. The issue is when I simply launch it without a master key like `./meilisearch` and I try to access other routes like stats indexes or try to add documents then I get an invalid API key error

![Screenshot 2023-05-10 151629](https://github.com/meilisearch/meilisearch/assets/101267639/606ef9c6-8cf3-4125-9a95-32c97121cb81)

**Comment by dureuill** [comment]
@oluademola 

It looks like the description of the issue is out of date as the issue isn't that Meilisearch starts with a wrong API key by default (but a background service).

Could you either update the description of this issue if there is a lingering issue, or close the issue?

**Comment by oluademola** [comment]
Hello @dureuill. I have updated the issue description. Thank you.

**Comment by oluademola** [comment]
Hello @dureuill. You are correct, I am very sorry for the false alarm. The issue was my env var. I thought I had removed it, but apparently, there are two types of env var in Windows. I had set a system env var, but I was only checking the list of the current user env var, and I couldn't find it; ends up it's being inheriting the system variable each time I launch meilisearch. Thanks for pointing this out.
